backend/src/main/java/gov/cdc/usds/simplereport/service/PatientSelfRegistrationLinkService.java

```diff
@@ -67,10 +81,29 @@ public String updateRegistrationLink(String link, String newLink) {
   public String updateRegistrationLink(String link, Boolean deleted) {
     PatientSelfRegistrationLink prl =
         prlrepo
-            .findByPatientRegistrationLink(link)
+            .findByPatientRegistrationLinkIgnoreCase(link)
             .orElseThrow(InvalidPatientSelfRegistrationLinkException::new);
     prl.setIsDeleted(deleted);
     prlrepo.save(prl);
     return prl.getLink();
   }
+
+  public String createRegistrationLink(Organization org) {
+    return createRegistrationLink(org, generateUniqueLink());
+  }
+
+  public String createRegistrationLink(Facility fac) {
+    return createRegistrationLink(fac, generateUniqueLink());
+  }
+
+  private String generateUniqueLink() {
+    String link = generateRandomLink();
+    return prlrepo.findByPatientRegistrationLinkIgnoreCase(link).isPresent()
+        ? generateRandomLink()
+        : link;
+  }
+
+  private static String generateRandomLink() {
+    return RandomStringUtils.random(LINK_LENGTH, "123456789abcdefghjkmnpqrstuvwxyz");
```

Comment:
**suggestion:** Let's also remove `1`, since it could be confused as `l` or `I` at first glance

Reply:
And: What do we think about `5` and `S`?

Reply:
👍  removed 1, 5, and S